Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/CoronaTest.kt

```diff
@@ -2,14 +2,29 @@ package de.rki.coronawarnapp.coronatest.type
 
 import com.google.gson.annotations.SerializedName
 import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestGUID
+import org.joda.time.Instant
 
 interface CoronaTest {
     val testGUID: CoronaTestGUID
-    val registrationToken: RegistrationToken
     val type: Type
-    val isRefreshing: Boolean
+
+    val registeredAt: Instant
+    val registrationToken: RegistrationToken
+
+    val isProcessing: Boolean
+    val lastError: Throwable?
```

Comment:
Do you need to persist those values?

Reply:
Not in storage, but in memory.
These reflect state and errors during polling (will rename name later when less conflict potential).
If the app dies and we are no longer in memory, then on next launch it's initialised again (when our app dies we are also no longer polling the server :grin:).

The error is used by the UI currently to show a dialog somewhere so "per app session" works ok here too. The subclasses should have this annotated with `@Transient`.